test/test_safemode_parser.rb

```diff
@@ -4,27 +4,32 @@ class TestSafemodeParser < Test::Unit::TestCase
   def test_vcall_should_be_jailed
     assert_jailed 'to_jail.a.to_jail.class', 'a.class'
   end
-    
+
   def test_call_should_be_jailed
     assert_jailed '(1.to_jail + 1).to_jail.class', '(1 + 1).class'
   end
-    
+
   def test_estr_should_be_jailed
     assert_jailed '"#{1.to_jail.class}"', '"#{1.class}"'
   end
-    
+
   def test_if_should_be_usable_for_erb
     assert_jailed "if true then\n 1\nend", "if true\n 1\n end"
   end
-    
+
   def test_if_else_should_be_usable_for_erb
     assert_jailed "if true then\n 1\n else\n2\nend", "if true\n 1\n else\n2\n end"
   end
-    
+
   def test_ternary_should_be_usable_for_erb
     assert_jailed "if true then\n 1\n else\n2\nend", "true ? 1 : 2"
   end
 
+  def test_kwargs_are_respected
+    # TODO: this is wrong for Ruby 3 and raises a deprecation warning in 2.7
+    assert_jailed "@article.to_jail.method_with_kwargs({ :a_keyword => true })", "@article.method_with_kwargs(a_keyword: true)"
```

Comment:
I tracked it down to this bit. Reading https://github.com/seattlerb/ruby2ruby/pull/55#issuecomment-1100342149 I'm afraid we may have to dive into ruby2ruby to properly deal with this.

Reply:
I should note that ruby2ruby 2.5.0 does add support for kwargs, so we can at least do it correctly.